src/detours.cpp

```diff
@@ -52,31 +52,89 @@ namespace rhost {
             char ch[1000];
             wcstombs(ch, lpText, _countof(ch));
             return HostMessageBox(ch, uType);
         }
 
         // MessageBoxA
-        decltype(MessageBoxW) *pMessageBoxA = NULL;
+        decltype(MessageBoxA) *pMessageBoxA = nullptr;
         int WINAPI DetourMessageBoxA(HWND hWnd, LPCSTR lpText, LPCSTR lpCaption, UINT uType) {
             return HostMessageBox(lpText, uType);
         }
 
+        std::unordered_map<HWND, std::string> hWndTitleMap;
+        decltype(CreateWindowExA) *pCreateWindowExA = nullptr;
+        HWND WINAPI DetourCreateWindowExA(DWORD dwExStyle, LPCSTR lpClassName, LPCSTR lpWindowName, DWORD dwStyle, int X, int Y, int nWidth, int nHeight, HWND hWndParent, HMENU hMenu, HINSTANCE hInstance, LPVOID lpParam) {
+            HWND hWnd = pCreateWindowExA(dwExStyle, lpClassName, lpWindowName, dwStyle, X, Y, nWidth, nHeight, hWndParent, hMenu, hInstance, lpParam);
+
+            // A window without parent or a window created to show message
+            size_t len = 0;
+            if (hWnd != nullptr && hWndParent == nullptr && 
+                hWndTitleMap.find(hWnd) == hWndTitleMap.end() && 
+                SUCCEEDED(StringCchLengthA(lpWindowName, STRSAFE_MAX_CCH, &len)) && len > 0) {
+
+                hWndTitleMap[hWnd] = std::string(lpWindowName);
+                std::string text = "Created window - " + hWndTitleMap[hWnd];
+
+                rhost::host::send_notification("!!", text);
+            }
+
+            return hWnd;
+        }
+
+        decltype(CreateWindowExW) *pCreateWindowExW = nullptr;
+        HWND WINAPI DetourCreateWindowExW(DWORD dwExStyle, LPCWSTR lpClassName, LPCWSTR lpWindowName, DWORD dwStyle, int X, int Y, int nWidth, int nHeight, HWND hWndParent, HMENU hMenu, HINSTANCE hInstance, LPVOID lpParam) {
+            HWND hWnd = pCreateWindowExW(dwExStyle, lpClassName, lpWindowName, dwStyle, X, Y, nWidth, nHeight, hWndParent, hMenu, hInstance, lpParam);
+            
+            // A window without parent or a window created to show message
+            size_t len = 0;
+            if (hWnd != nullptr && hWndParent == nullptr && 
+                hWndTitleMap.find(hWnd) == hWndTitleMap.end() && 
+                SUCCEEDED(StringCchLengthW(lpWindowName, STRSAFE_MAX_CCH, &len) && len > 0)) {
+
+                std::wstring windowTitle = std::wstring(lpWindowName);
+                std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t> strConverter;
+
+                hWndTitleMap[hWnd] = strConverter.to_bytes(windowTitle);
+                std::string text = "Created window - " + hWndTitleMap[hWnd];
+
+                rhost::host::send_notification("!!", text);
+            }
+
+            return hWnd;
+        }
+
+        decltype(DestroyWindow) *pDestroyWindow = nullptr;
+        BOOL WINAPI DetourDestroyWindow(HWND hWnd) {
+            BOOL stat = pDestroyWindow(hWnd);
+            if (hWndTitleMap.find(hWnd) != hWndTitleMap.end()) {
```

Comment:
The usual pattern is to save the iterator returned by `find`, so that you don't have to do the second lookup with `[]`, but just use the iterator to access the value via `iter->second`. You can then also use it for `erase`.